mmv1/templates/terraform/examples/secure_source_manager_instance_private_psc.tf.erb

```diff
@@ -0,0 +1,249 @@
+}
+
+resource "google_compute_network" "network2" {
+  name = "<%= ctx[:vars]['network2_id'] %>"
+  auto_create_subnetworks = false
+}
+
+resource "google_compute_subnetwork" "subnet2" {
+  name = "<%= ctx[:vars]['subnet2_id'] %>"
+  region = "us-central1"
+  network = google_compute_network.network2.id
+  ip_cidr_range = "10.0.60.0/24"
+  private_ip_google_access = true # needed for "gcloud auth print-access-token" to work
+}
+
+resource "google_compute_address" "address" {
+  name = "<%= ctx[:vars]['address_id'] %>"
+  region = "us-central1"
+  address = "10.0.60.100"
+  address_type = "INTERNAL"
+  subnetwork = google_compute_subnetwork.subnet2.id
+}
+
+resource "google_compute_forwarding_rule" "fw_rule_service_attachment" {
+  name = "<%= ctx[:vars]['fw_rule_to_service_attachment_id'] %>"
+  region = "us-central1"
+
+  load_balancing_scheme = ""
+  ip_address = google_compute_address.address.id
+  network = google_compute_network.network2.id
+
+  target = google_compute_service_attachment.service_attachment.id
+}
+
+resource "google_dns_managed_zone" "private_zone" {
+  name = "<%= ctx[:vars]['dns_zone_id'] %>"
+  dns_name = "p.sourcemanager.dev."
+  visibility = "private"
+  private_visibility_config {
+    networks {
+      network_url = google_compute_network.network2.id
+    }
+  }
+}
+
+resource "google_dns_record_set" "ssm_instance_html_record" {
+  name = "${google_secure_source_manager_instance.<%= ctx[:primary_resource_id] %>.host_config.0.html}."
+  type = "A"
+  ttl = 300
+  managed_zone = google_dns_managed_zone.private_zone.name
+  rrdatas = [google_compute_forwarding_rule.fw_rule_service_attachment.ip_address]
+}
+
+resource "google_dns_record_set" "ssm_instance_api_record" {
+  name = "${google_secure_source_manager_instance.<%= ctx[:primary_resource_id] %>.host_config.0.api}."
+  type = "A"
+  ttl = 300
+  managed_zone = google_dns_managed_zone.private_zone.name
+  rrdatas = [google_compute_forwarding_rule.fw_rule_service_attachment.ip_address]
+}
+
+resource "google_dns_record_set" "ssm_instance_git_record" {
+  name = "${google_secure_source_manager_instance.<%= ctx[:primary_resource_id] %>.host_config.0.git_http}."
+  type = "A"
+  ttl = 300
+  managed_zone = google_dns_managed_zone.private_zone.name
+  rrdatas = [google_compute_forwarding_rule.fw_rule_service_attachment.ip_address]
+}
+
+// Follow https://cloud.google.com/secure-source-manager/docs/create-private-service-connect-instance#root-ca-api
+resource "google_compute_instance" "compute_instance" {
+  name = "<%= ctx[:vars]['compute_instance_id'] %>"
+  machine_type = "n2-standard-2"
+  zone         = "us-central1-a"
+
+  boot_disk {
+    initialize_params {
+      image = "debian-cloud/debian-11"
+    }
+  }
+  scratch_disk {
+    interface = "NVME"
+  }
+  network_interface {
+    network = google_compute_network.network2.id
+    subnetwork = google_compute_subnetwork.subnet2.id
+  }
+}
+
+resource "google_privateca_ca_pool" "ca_pool" {
```

Comment:
ca_pool and related resource should happen before creating ssm instance.

Reply:
Isn't this reflected in the dependency of the ssm instance, and it using the ca_pool?